extensions/amp-form/0.1/amp-form.js

```diff
@@ -880,27 +848,39 @@ export class AmpForm {
   }
 
   /**
-   * Transition the form the the submit error state.
-   * @param {!Error} error
+   * @param {*} e
```

Comment:
Can you help me understand why `{*}` is preferred?

Reply:
It's convention that promises be rejected with `Error` objects, but rejection reasons can be anything. Dug around a bit and I don't think that Closure has a way of specifying this via type annotation.

I think casting to `Error` is error prone since it gives a false sense of type safety. And if we do need to typecast, it's more obvious what we're assuming if the typecast happens inside the function.